common/utils/utils.go

```diff
@@ -500,6 +500,63 @@ func ReadSpannerSchema(ctx context.Context, conv *internal.Conv, client *sp.Clie
 	return nil
 }
 
+// CompareSchema compares the spanner schema of two conv objects and returns specific error if they don't match
+func CompareSchema(conv1, conv2 *internal.Conv) error {
```

Comment:
looks like this function does not handle cases when the lists/map are differently ordered.

Reply:
Also what about cases when the session file contents are a subset of the spanner schema contents? We might want to handle that case, where the user added some extra tables but used HB to only load a partial set. It shouldn't be too much of a change on our side but will improve usability.

Reply:
For tables and columns we are reading from map so ordering won't affect and for indexes we store them in the order of definition.

Reply:
Added handling for case where spanner database may have extra table. But in case of same tables we should check for similarity in definition to avoid data write failures.